openstack/datadog_checks/openstack/openstack.py

```diff
@@ -844,24 +858,64 @@ def get_stats_for_single_hypervisor(self, hyp_id, instance, host_tags=None):
             for i, avg in enumerate([1, 5, 15]):
                 self.gauge('openstack.nova.hypervisor_load.{0}'.format(avg), load_averages[i], tags=tags)
 
-    def get_all_server_ids(self, filter_by_host=None):
+    # Get all of the server IDs and their metadata and cache them
+    # After the first run, we will only get servers that have changed state since the last collection run
+    def get_all_servers(self, i_key, filter_by_host=None):
         query_params = {}
         if filter_by_host:
             query_params["host"] = filter_by_host
 
-        url = '{0}/servers'.format(self.get_nova_endpoint())
+        # If we don't have a timestamp for this instance, default to None
+        if i_key not in self.changes_since_time:
+            self.changes_since_time[i_key] = None
+
+        if self.changes_since_time[i_key]:
+            query_params['changes-since'] = self.changes_since_time[i_key]
```

Comment:
You can remove these 4 lines and just do:

`query_params['changes-since'] = self.changes_since_time.get(i_key)`